components/ConvertSteps/ConvertSteps.js

```diff
@@ -1,178 +1,95 @@
-import React, { useCallback, useEffect, useState } from 'react'
+import React, { useCallback, useState } from 'react'
 import Divider from './Divider'
 import PropTypes from 'prop-types'
 import StepperLayout from './StepperLayout'
-import { useAllowance, useOpenOrder, useClaimOrder } from 'lib/web3-contracts'
-import Step from './Step'
+import ManageStep from './ManageStep'
 import StepperTitle from './StepperTitle'
 
-function ConvertSteps({ toAnj, amountSource, amountRecipient, onReturnHome }) {
-  const changeAllowance = useAllowance()
-  const openOrder = useOpenOrder()
-  const claimOrder = useClaimOrder()
+function ConvertSteps({ toAnj, fromAmount, toAmount, onReturnHome, steps }) {
+  const [stepperStatus, setStepperStatus] = useState('working')
+  const [stepperStage, setStepperStage] = useState(0)
+  const [retryStep, setRetryStep] = useState(null)
 
-  const [stepperStage, setStepperStage] = useState('working')
-  const [currentStep, setCurrentStep] = useState(
-    toAnj ? 'approval' : 'buyOrder'
-  )
-  const [stepState, setStepState] = useState({
-    approval: {
-      status: 'waiting',
-      active: true,
-    },
-    buyOrder: {
-      status: 'waiting',
-      active: toAnj ? false : true,
-      hash: '',
-    },
-    claimOrder: {
-      status: 'waiting',
-      active: false,
-      hash: '',
-    },
-  })
-
-  const applyStepState = useCallback((step, status, hash) => {
-    setStepState(prevState => {
-      return {
-        ...prevState,
-        [step]: {
-          status: status,
-          active: true,
-          hash: hash ? hash : prevState[step].hash,
-        },
-      }
-    })
-  }, [])
-
-  const handleClaimOrderStep = useCallback(
-    async buyOrderHash => {
-      const stepType = 'claimOrder'
-
-      setCurrentStep(stepType)
+  function handleRetry() {
+    setStepperStatus('working')
+    setRetryStep(stepperStage)
+  }
 
-      try {
-        // Awaiting confirmation
-        applyStepState(stepType, 'waiting')
-        const transaction = await claimOrder(buyOrderHash, toAnj)
+  const clearRetry = () => setRetryStep(null)
 
-        // Mining transaction
-        applyStepState(stepType, 'working', transaction.hash)
-        await transaction.wait()
+  const handleError = () => {
+    clearRetry()
```

Comment:
Sanity checking: would this not let the user retry a step if there was an error?

Reply:
The function of this isn't particularly clear but its purpose is to flag that the step is "retrying" so that it doesn't attempt to run the signing code more than once after triggering a retry.

I'll update the code to make this clearer but I think much of what's happening in this file would benefit from a raising of logic as you suggested in a later comment.